server/controllers/auth/signup.js

```diff
@@ -0,0 +1,65 @@
+const { hash } = require('bcryptjs');
+const { sign } = require('jsonwebtoken');
+const { addUserQuery } = require('../../database/queries/users');
+
+const signUp = (req, res) => {
+  const { username, password, email } = req.body;
+  // hash password
+  hash(password, 10, (err, hashedPassword) => {
+    if (err) {
+      res.status(500).json({
+        msg: 'Internal Server Error',
+        status: 500,
+      });
+    } else {
+      // add user to the database
+      addUserQuery({
+        username,
+        email,
+        password: hashedPassword,
+      })
+        .then((data) => data.rows[0])
+        .then((user) => {
+          sign(
```

Comment:
convert sign here to promise manually as `jsonwebtoken` doesn't support promises, export it from utils and use it here